app/client/ui-modules/role-dashboard/ProfilePage.tsx

```diff
@@ -0,0 +1,174 @@
+                                }
+                                onImageChange={(file) => {
+                                    const imageUrl = URL.createObjectURL(file);
+                                    setProfileImage(imageUrl);
+                                }}
+                            />
+
+                            <div className="mt-[10px]">
+                                <h2 className="text-xl font-bold">
+                                    {profile.firstName} {profile.lastName}
+                                </h2>
+                                <p className="text-sm text-muted-foreground">
+                                    Agent since ...
+                                </p>
+                                <div className="flex gap-2 flex-wrap">
+                                    <span className="text-xs bg-white text-black px-2 py-1 rounded-full border border-gray-400">
+                                        Verified Agent{" "}
+                                        {/*TODO: add switch statements to change nav bars based on users role i.e agent  */}
+                                    </span>
+                                </div>
+                            </div>
+                        </div>
+                        <div className="flex gap-2 mt-1">
+                            {isEditing ? (
+                                <>
+                                    <Button
+                                        onClick={() =>
+                                            dispatch(setEditing(false))
+                                        }
+                                        className="w-32 mt-1 hover:bg-gray-300 cursor-pointer transition"
+                                    >
+                                        Cancel Edit
+                                    </Button>
+                                    <Button
+                                        onClick={handleSave}
+                                        className="w-32 mt-1 hover:bg-gray-300 cursor-pointer transition"
+                                    >
+                                        Save Profile
+                                    </Button>
+                                </>
+                            ) : (
+                                <Button
+                                    onClick={() => dispatch(setEditing(true))}
+                                    className="w-32 mt-1 hover:bg-gray-300 cursor-pointer transition"
+                                >
+                                    Edit Profile
+                                </Button>
+                            )}
+                        </div>
+                    </div>
+
+                    <div className="grid grid-cols-1 sm:grid-cols-2 gap-6 mb-6">
+                        <div className="col-span-full flex justify-end"></div>
+                        <PersonalInfoCard
```

Comment:
this is just my personal preferences but instead of repeating the same thing for each card you could use the react mapping structure to avoid duplications.